frontend/public/components/catalog/hooks/useCatalogCategories.ts

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import { safeLoad } from 'js-yaml';
+
+import { ConfigMapKind } from '@console/internal/module/k8s';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+
+import { defaultCatalogCategories } from '../utils/default-categories';
+
+interface DeveloperConsoleCatalogCategory {
+  id: string;
+  label: string;
+  tags: string[];
+  subcategories: DeveloperConsoleCatalogSubcategory[];
+}
+
+interface DeveloperConsoleCatalogSubcategory {
+  id: string;
+  label: string;
+  tags: string[];
+}
+
+interface ConsoleConfig {
+  customization: {
+    developerCatalog: {
+      categories: DeveloperConsoleCatalogCategory[];
+    };
+  };
+}
+
+export const useCatalogCategories = () => {
+  const [configMap, loaded, loadError] = useK8sWatchResource<ConfigMapKind>({
+    kind: 'ConfigMap',
+    namespace: 'openshift-console',
+    name: 'console-config',
+  });
```

Comment:
The console frontend shouldn't be fetching this config map directly. Normal users won't have access. The right way to do this is to update the console-operator to watch the custom categories config map and roll out a new console revision when it changes, wiring through updated config to the console backend.

cc @jhadvig

Reply:
Oh, okay. Just to verify if I get the right way now, which I just discussed with @christianvogt:

Instead of reading the `ConfigMap` via k8 watch, extract the `customization.developerCatalog.categories` as SERVER_FLAG "developerCatalog.categories" in [pkg/serverconfig/config.go](https://github.com/openshift/console/blob/43e041787cf3899cf6ec4c6ea5e279e37b64c1c6/pkg/serverconfig/config.go#L165-L181). Then it will be automatically available as `windows.SERVER_FLAG.developerCatalog.categories`

And then I just need to return the server flag || defaultCatalogCategories in the hook?

Reply:
Right, the backend sets the server flag with the categories. We might not need a hook at all since the data won't change after console load.